features/shopify-markets-js-api.md

```diff
@@ -0,0 +1,340 @@
+            listPriceElement: '.nosto-product-price > .nosto-newprice.money',
+            descriptionElement: 'x-nosto-desc',
+        })
+```
+
+With only the necessary parameters as below:
+
+```javascript
+_targetWindow.Nosto.migrateToShopifyMarket({
+            productSectionElement: '.nosto-product-info',
+            productUrlElement: '.nosto-product-url',
+            productHandleAttribute: 'x-nosto-handle',
+            priceElement: '.nosto-product-price > .money',
+        })
+```
+
+##### Using API with variant options
+
+{% hint style="info" %} 
+Only the necessary part of the template is shown below. It's not a complete recommendation template.
+While loading the variant options, is not necessary to load the values for the sub-variants. The API loads the data automatically for the selected market and laguage
+{% endhint %}
+
+```html
+<div class="nosto-product-url nosto-sm-product-info" x-nosto-handle="$!product.lastPathOfProductUrl()">
+    <span class="x-nosto-title">$!product.name</span>
+    <span class="x-nosto-desc">$!product.description</span>
+    <span class="x-nosto-title">$!product.name</span>
+
+    <a class="nosto-product-url" href="$!product.url" alt="$!product.name">
+        <div class="nosto-image-container img-$imgCount"></div>
+    </a>
+
+    <a href="$!product.url" class="nosto-product-brand">$!product.brand.truncate(100)</a>
+    <a href="$!product.url" class="nosto-product-url nosto-product-name">$!product.name.truncate(100)</a>
+    <a href="$!product.url" class="nosto-product-url nosto-product-price">
+        #if($product.discounted)
+            <span class="nosto-oldprice money">$!product.listPrice</span>
+        #end
+        <span class="nosto-newprice money">$!product.price</span>
+    </a>
+    <div class="nosto-rating">
+        #set($ratingInt = $math.round($!product.ratingValue.getAsNumber()).intValue())
+        #set($ratingRest = 5 - $ratingInt)
+        #if($ratingInt > 0)
+            #foreach ($number in [1..$ratingInt])
+                <span class="nosto-star-full">★</span>
+            #end
+        #end
+
+        #if($ratingRest != 0)
+            #foreach ($number in [1..$ratingRest])
+                <span class="nosto-star-empty">★</span>
+            #end
+        #end
+
+        <span class="nosto-reviews">($!product.reviewCount)</span>
+    </div>
+    
+    <div class="variant-options">
+        <h3 class="size-option-title">Size</h3>
+        <div class="nosto-sizes">
+            #foreach($sku in $product.skus)
+                <span x-nosto-product="$!product.productId" class="nosto-size #if(!$!sku.available) nosto-disabled #end">$!sku.getCustomField('Size')</span>
+            #end
+        </div>
+        <h3 class="color-option-title">Color</h3>
+        <div class="nosto-colors">
+        </div>
+    </div>
+</div>
+```
+
+For the template above, you can call the JS API  
+
+```javascript
+(function() {
```

Comment:
None of this JS is related to the Markets workaround, is it? I think adding this code to the example could cause confusion.

Reply:
It's in fact about the JS around. The script provides an example of how the logic works when a value of variation dropdown changes.